### PR TITLE
Bench improvement

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
Switch bencher to criterion. Update sum_n to do work instead of running pre-compiled code

### CODE REVIEW

File: runtime/near-vm-runner/tests/test-contract-rs/src/lib.rs [hunk 1]
```diff
@@ -370,12 +333,13 @@ pub unsafe fn sum_n() {
...
     let n = u64::from_le_bytes(data);
 
-    let mut sum = 0u64;
-    for i in 0..n {
-        sum += i;
+    let mut sum = 0u128;
+    for i in 1..n + 1 {
+        sum += (i * i) as u128;
     }
+    sum /= n as u128;
```
Comment (nearmax) [inline runtime/near-vm-runner/tests/test-contract-rs/src/lib.rs]: Why this line needed?

File: runtime/near-vm-runner/benches/bench.rs [hunk 2]
```diff
@@ -52,68 +53,96 @@ fn assert_run_result((outcome, err): (Option<VMOutcome>, Option<VMError>), expec
...
+            group.sample_size(20);
+        }
+        let (mut external, context, config, promise_results, code) = setup(&[n]);
+        let benchmark_param_display = format!("n={}", n);
+        let mut expected_value = 0u128;
+        for i in 1..n + 1 {
+            expected_value += (i * i) as u128;
+        }
+        expected_value /= n as u128;
+        group.bench_function(BenchmarkId::new("wasm", benchmark_param_display.clone()), |b| {
```
Comment (nearmax) [inline runtime/near-vm-runner/benches/bench.rs]: benchmark id should include name of the function, e.g. "sum_n"?

Comment (evgenykuzyakov) [reply runtime/near-vm-runner/benches/bench.rs]: the group has the name

File: runtime/near-vm-runner/tests/test-contract-rs/src/lib.rs (hunk 1 above)
Comment (evgenykuzyakov) [reply runtime/near-vm-runner/tests/test-contract-rs/src/lib.rs]: Otherwise it overfits `u64` for larger sums

**Comment by nearmax** [comment]
@evgenykuzyakov Is this an abandoned PR? Should we close it?

**Comment by evgenykuzyakov** [comment]
Sorry. It still worth to complete it. I just didn't get back to address the changes. Will take a look later this week.

**Comment by nearmax** [comment]
@evgenykuzyakov Did you have a chance to look at it?

**Comment by ilblackdragon** [comment]
@evgenykuzyakov ping?

**Comment by nearmax** [comment]
We can probably close it since now we have `runtime-params-estimator`.

**Comment by evgenykuzyakov** [comment]
It's also less useful, since it just runs wasm in isolation.

**Pull request closed without merging**